vmdb/app/controllers/ems_common.rb

```diff
@@ -598,17 +598,25 @@ def set_form_vars
     @edit[:new][:emstype] = @ems.emstype
     @edit[:amazon_regions] = get_amazon_regions if @ems.kind_of?(EmsAmazon)
     @edit[:new][:port] = @ems.port
+    @edit[:new][:provider_id] = @ems.provider_id
     if @ems.zone.nil? || @ems.my_zone == ""
       @edit[:new][:zone] = "default"
     else
       @edit[:new][:zone] = @ems.my_zone
     end
-    @edit[:server_zones] = Array.new
+    @edit[:server_zones] = []
     zones = Zone.order('lower(description)')
     zones.each do |zone|
       @edit[:server_zones].push([zone.description, zone.name])
     end
 
+    @edit[:openstack_infra_providers] = []
+    openstack_infra_providers = ProviderOpenstack.order('lower(name)')
+    @edit[:openstack_infra_providers].push(["---", nil])
+    openstack_infra_providers.each do |openstack_infra_provider|
```

Comment:
@blomquisg : We'll deal with that with @Ladas, I believe. Generally it would be *nice* if you make UI code better by your changes and making more smaller methods with nice names if much better than matching the spaghetti style we have on so many places ;-)